internal/pkg/provision/check/default.go

```diff
@@ -0,0 +1,61 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package check
+
+import (
+	"context"
+	"time"
+
+	"github.com/talos-systems/talos/internal/pkg/conditions"
+	"github.com/talos-systems/talos/internal/pkg/provision"
+)
+
+// DefaultClusterChecks returns a set of default Talos cluster readiness checks.
+func DefaultClusterChecks() []ClusterCheck {
+	return []ClusterCheck{
```

Comment:
this is the core of the new checks: list of all the checks to execute against the cluster